From 749be35e67e8026991d57deae402c2092839d8f2 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 22 Nov 2023 16:59:44 +0100
Subject: [PATCH 1/4] deps/wxWidgets: Download wxwidgets as an archive

Instead of checking out the git repo, use the archive for the release
which will make it possible to build BambuStudio offline.
---
 deps/wxWidgets/wxWidgets.cmake | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/deps/wxWidgets/wxWidgets.cmake b/deps/wxWidgets/wxWidgets.cmake
index 227abe18539f..46f676c019ef 100644
--- a/deps/wxWidgets/wxWidgets.cmake
+++ b/deps/wxWidgets/wxWidgets.cmake
@@ -1,4 +1,5 @@
-set(_wx_git_tag v3.1.5)
+set(_wx_version 3.1.5)
+set(_wx_sha256 bc5b81a54208c8d7f11a19e9641ea12b870ae473ec5f5a99fd9df57c0558e4a3)
 
 set(_wx_toolkit "")
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
@@ -25,8 +26,8 @@ else ()
 endif ()
 
 bambustudio_add_cmake_project(wxWidgets
-    GIT_REPOSITORY "https://github.com/wxWidgets/wxWidgets"
-    GIT_TAG ${_wx_git_tag}
+    URL https://github.com/wxWidgets/wxWidgets/releases/download/v${_wx_version}/wxWidgets-${_wx_version}.zip
+    URL_HASH SHA256=${_wx_sha256}
     PATCH_COMMAND ${_patch_cmd}
     DEPENDS ${PNG_PKG} ${ZLIB_PKG} ${EXPAT_PKG} dep_TIFF dep_JPEG
     CMAKE_ARGS
-- 
2.43.0


From 6c44e70e9e53be23f5b48a6206e6a959a94732ce Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sun, 10 Dec 2023 12:35:30 +0100
Subject: [PATCH 2/4] deps/wxWidgets: Upgrade to 3.2.4

---
 deps/wxWidgets/wxWidgets.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/deps/wxWidgets/wxWidgets.cmake b/deps/wxWidgets/wxWidgets.cmake
index 46f676c019ef..229c862cf853 100644
--- a/deps/wxWidgets/wxWidgets.cmake
+++ b/deps/wxWidgets/wxWidgets.cmake
@@ -1,5 +1,5 @@
-set(_wx_version 3.1.5)
-set(_wx_sha256 bc5b81a54208c8d7f11a19e9641ea12b870ae473ec5f5a99fd9df57c0558e4a3)
+set(_wx_version 3.2.4)
+set(_wx_sha256 051aa9d5a1b39fec940efd963b702a6082d5eb57935245edcbab21dfe19be209)
 
 set(_wx_toolkit "")
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-- 
2.43.0


From 950d5dfc8702261c2912cd1df45408c60a5edde9 Mon Sep 17 00:00:00 2001
From: Gregor Riepl <onitake@gmail.com>
Date: Mon, 27 Nov 2023 13:01:55 +0100
Subject: [PATCH 3/4] Make initializers explicit to avoid ambiguous
 wxArrayString overloads
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/slic3r/GUI/PhysicalPrinterDialog.cpp: In member function ‘void Slic3r::GUI::PhysicalPrinterDialog::build_printhost_settings(Slic3r::GUI::ConfigOptionsGroup*)’:
src/slic3r/GUI/PhysicalPrinterDialog.cpp:332: error: call of overloaded ‘set_values(<brace-enclosed initializer list>)’ is ambiguous
  332 |         choice->set_values({ m_config->opt_string("printhost_port") });
      |
---
 src/slic3r/GUI/PhysicalPrinterDialog.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/slic3r/GUI/PhysicalPrinterDialog.cpp b/src/slic3r/GUI/PhysicalPrinterDialog.cpp
index 4d6d48a26e52..1ab467214475 100644
--- a/src/slic3r/GUI/PhysicalPrinterDialog.cpp
+++ b/src/slic3r/GUI/PhysicalPrinterDialog.cpp
@@ -329,7 +329,7 @@ void PhysicalPrinterDialog::build_printhost_settings(ConfigOptionsGroup* m_optgr
     // Always fill in the "printhost_port" combo box from the config and select it.
     {
         Choice* choice = dynamic_cast<Choice*>(m_optgroup->get_field("printhost_port"));
-        choice->set_values({ m_config->opt_string("printhost_port") });
+        choice->set_values(std::vector<std::string>({ m_config->opt_string("printhost_port") }));
         choice->set_selection();
     }
 
-- 
2.43.0


From 342c8af9e0dcd6e06b2b2f7a35e19a29a48628cc Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sun, 10 Dec 2023 15:46:28 +0100
Subject: [PATCH 4/4] Use system PCRE library

PCRE is a common part of the OS.
---
 deps/wxWidgets/wxWidgets.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deps/wxWidgets/wxWidgets.cmake b/deps/wxWidgets/wxWidgets.cmake
index 229c862cf853..c8e0045708ae 100644
--- a/deps/wxWidgets/wxWidgets.cmake
+++ b/deps/wxWidgets/wxWidgets.cmake
@@ -45,7 +45,7 @@ bambustudio_add_cmake_project(wxWidgets
         -DwxUSE_WEBVIEW=ON
         ${_wx_edge}
         -DwxUSE_WEBVIEW_IE=OFF
-        -DwxUSE_REGEX=builtin
+        -DwxUSE_REGEX=sys
         -DwxUSE_LIBXPM=builtin
         -DwxUSE_LIBSDL=OFF
         -DwxUSE_XTEST=OFF
-- 
2.43.0

